### DISCUSSION TITLE
https update firmware w5500

### DISCUSSION THREAD

**Discussion by mikeme2009** [JAndrassy/ArduinoOTA#140] [General]
Hi

Is it possible with your library to update the esp32 firmware via https and w5500 shield but without using sdcard?
Thanks

**Comment by JAndrassy** [comment]
I guess it is possible but it will be a Frankenstein's monster, because you would have to use the SSLClient library over the Ethernet library.
for Ethernet on esp32 use the native Ethernet support of the esp32 in the WiFi library with one of the supported Ethernet Phy modules.

**Comment by mikeme2009** [comment]
I started my project using esp32 and Lan8710A. Everything was going well but there are no more stocks of Lan8710a (only in 2023) and I had to find an alternative. Now I have esp32 and w5500. Can I use your library and update the firmware directly from the server via http?

**Reply by JAndrassy** [reply]
> Can I use your library and update the firmware directly from the server via http?

for that you can use the esp32 Update library directly

**Reply by BobLynas** [reply]
HI Mike, I would also love to update the ESP32 firmware, from a served webpage, via a W5500 module. In fact i would love to know how to just pass a big file from a local webpage to the ESP32 for things other than program updates :)

**Comment by mikeme2009** [comment]
I tried but got an error... 
Starting OTA

Please Wait it takeassertion "Invalid mbox" failed: file "/home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/lwip/lwip/src/api/tcpip.c", line 267, function: tcpip_callback_with_block

abort() was called at PC 0x4011008b on core 0

**Reply by JAndrassy** [reply]
that is the 'fun' with the esp32 or more precisely with the Espressif IDF
